### ISSUE TITLE
ControlofIslandIsMissing

### ISSUE THREAD

**Issue by VeikoAunapuu** [Baltic-RCC/EMF#122]

**Comment by Haigutus** [comment]
Removal of small islands might work

```python
    # Remove islands smaller than SMALL_ISLAND_SIZE
    SMALL_ISLAND_SIZE = 3
    small_islands = pandas.DataFrame(solved_data.query("KEY == 'TopologicalIsland.TopologicalNodes'").ID.value_counts()).reset_index().query("count <= @SMALL_ISLAND_SIZE")
    solved_data = triplets.rdf_parser.remove_triplet_from_triplet(solved_data, small_islands, columns=["ID"])
    logger.info(f"Removed {len(small_islands)} island(s) with size <= {SMALL_ISLAND_SIZE}")
```

or setting island reference node

```python
    # Get largest island ID
    largest_island_id = data.query("KEY == 'TopologicalIsland.TopologicalNodes'").ID.value_counts().index[0]

    # Set island reference node
    slack_machine_terminal_id = data.query("VALUE == @slack_machine.ID and KEY == 'Terminal.ConductingEquipment'").iloc[0].ID
    slack_machine_TN_id = data.query("ID == @slack_machine_terminal_id and KEY == 'Terminal.TopologicalNode'").iloc[0].VALUE

    logger.info(f"Setting largest TopologicalIsland [{largest_island_id}] AngleRefTopologicalNode to {slack_machine_TN_id}")
    data.set_VALUE_at_KEY_and_ID(id=largest_island_id, key='TopologicalIsland.AngleRefTopologicalNode', value=slack_machine_TN_id)

**Comment by VeikoAunapuu** [comment]
Current Island error has 5 topological nodes, SMALL_ISLAND_SIZE should be at least 5

**Comment by Haigutus** [comment]
@gpoderys did you commit the solution?

**Comment by Haigutus** [comment]
quick fix committed, root cause need investigation

**Comment by Haigutus** [comment]
Possibly caused by SV load value taken from SSH in case not solved island, related to #125, closing for now
